src/dual_equality_constraints.jl

```diff
@@ -0,0 +1,112 @@
+function add_dual_equality_constraints(dual_model::AbstractModel{T}, primal_model::AbstractModel{T},
```

Comment:
I was using AbstractModel because I don't really know what is the difference of implementation to extend to all MOI.ModelLike.

Reply:
I am having some trouble to define the type of the dual model, for now, I wrote
```julia
    T = Float64
    dual_model = DualizableModel{T}()
```
so I can continue my tests.

But it would be nice to know `T` from the primal model in a natural way. I thought of some ideas that I consider terrible solutions.

Reply:
The user should give you the `model`. Later this will be the solver so it's better if the function gets the model as input.

> I was using AbstractModel because I don't really know what is the difference of implementation to extend to all MOI.ModelLike.

The difference is: you cannot access internal fields and should still to the MOI API :-P

Reply:
hmm, I think this change lots of things, you mean that I should not do `model.moi_scalaraffinefunction.moi_greaterthan` to access the information of these kinds of constraints but use MOI instead?

Reply:
This would completely change the `utils.jl` file, all other files would have minor updates

Reply:
>  you mean that I should not do `model.moi_scalaraffinefunction.moi_greaterthan` to access the information of these kinds of constraints but use MOI instead?

Exactly, you should use `MOI.ListOfConstraints` and `MOI.ListOfConstraintIndices`

Reply:
It simplified a lot of things! Thank you :)